# Data Visualization with Matplotlib
_Week 4_



# Exercise 1. Loading and Inspecting Data:

1. Load the dataset into a Pandas DataFrame.

2. Display the first 5 rows of the DataFrame.

3. Drop the column called `target`.

4. Show a summary of the DataFrame, including column data types and non-null values.

NOTE:

In this dataset, besides null values, we have some columns that have '?' in the file. There is a way to replace a string with a null value. Check the parameters of the read_csv function to use the appropriate parameter for this purpose.

In [5]:
import pandas as pd
columns = ['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach', 'exang', 'oldpeak', 'slope', 'ca', 'thal', 'target']
df = pd.read_csv('processed.cleveland.data', names=columns) 

In [6]:
df.head(5)

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0  63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1  67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2  67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3  37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4  41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   

   slope   ca thal  target  
0    3.0  0.0  6.0       0  
1    2.0  3.0  3.0       2  
2    2.0  2.0  7.0       1  
3    3.0  0.0  3.0       0  
4    1.0  0.0  3.0       0

In [8]:
df.drop(columns='target')

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0    63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1    67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2    67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3    37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4    41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   
..    ...  ...  ...       ...    ...  ...      ...      ...    ...      ...   
298  45.0  1.0  1.0     110.0  264.0  0.0      0.0    132.0    0.0      1.2   
299  68.0  1.0  4.0     144.0  193.0  1.0      0.0    141.0    0.0      3.4   
300  57.0  1.0  4.0     130.0  131.0  0.0      0.0    115.0    1.0      1.2   
301  57.0  0.0  2.0     130.0  236.0  0.0      2.0    174.0    0.0      0.0   
302  38.0  1.0  3.0     138.0  175.0  0.0      0.0    173.0    0.0      0.0   

     slope   ca thal  
0      3.0  0.0  6.0  
1      2.0  3.0  3.0  
2      2.0  2.0  7.0  
3      3.0  0.0  3.0  
4      1.0  0.0  3.0  
..     ...  ...  ...  
298    2.0  0.0  7.0  
299    2.0  2.0  7.0  
300    2.0  1.0  7.0  
301    2.0  1.0  3.0  
302    1.0    ?  3.0  

[303 rows x 13 columns]

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    float64
 1   sex       303 non-null    float64
 2   cp        303 non-null    float64
 3   trestbps  303 non-null    float64
 4   chol      303 non-null    float64
 5   fbs       303 non-null    float64
 6   restecg   303 non-null    float64
 7   thalach   303 non-null    float64
 8   exang     303 non-null    float64
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    float64
 11  ca        303 non-null    object 
 12  thal      303 non-null    object 
 13  target    303 non-null    int64  
dtypes: float64(11), int64(1), object(2)
memory usage: 33.3+ KB


# Exercise 2: Data Cleaning
1. Drop the rows with missing values.

2. Drop the duplicated rows.

In [ ]:
df.dropna(inplace=True)

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0    63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1    67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2    67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3    37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4    41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   
..    ...  ...  ...       ...    ...  ...      ...      ...    ...      ...   
298  45.0  1.0  1.0     110.0  264.0  0.0      0.0    132.0    0.0      1.2   
299  68.0  1.0  4.0     144.0  193.0  1.0      0.0    141.0    0.0      3.4   
300  57.0  1.0  4.0     130.0  131.0  0.0      0.0    115.0    1.0      1.2   
301  57.0  0.0  2.0     130.0  236.0  0.0      2.0    174.0    0.0      0.0   
302  38.0  1.0  3.0     138.0  175.0  0.0      0.0    173.0    0.0      0.0   

     slope   ca thal  target  
0      3.0  0.0  6.0       0  
1      2.0  3.0  3.0       2  
2      2.0  2.0  7.0       1  
3      3.0  0.0  3.0       0  
4      1.0  0.0  3.0       0  
..     ...  ...  ...     ...  
298    2.0  0.0  7.0       1  
299    2.0  2.0  7.0       2  
300    2.0  1.0  7.0       3  
301    2.0  1.0  3.0       1  
302    1.0    ?  3.0       0  

[303 rows x 14 columns]

In [12]:
df.drop_duplicates(inplace=True)
df

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0    63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1    67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2    67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3    37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4    41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   
..    ...  ...  ...       ...    ...  ...      ...      ...    ...      ...   
298  45.0  1.0  1.0     110.0  264.0  0.0      0.0    132.0    0.0      1.2   
299  68.0  1.0  4.0     144.0  193.0  1.0      0.0    141.0    0.0      3.4   
300  57.0  1.0  4.0     130.0  131.0  0.0      0.0    115.0    1.0      1.2   
301  57.0  0.0  2.0     130.0  236.0  0.0      2.0    174.0    0.0      0.0   
302  38.0  1.0  3.0     138.0  175.0  0.0      0.0    173.0    0.0      0.0   

     slope   ca thal  target  
0      3.0  0.0  6.0       0  
1      2.0  3.0  3.0       2  
2      2.0  2.0  7.0       1  
3      3.0  0.0  3.0       0  
4      1.0  0.0  3.0       0  
..     ...  ...  ...     ...  
298    2.0  0.0  7.0       1  
299    2.0  2.0  7.0       2  
300    2.0  1.0  7.0       3  
301    2.0  1.0  3.0       1  
302    1.0    ?  3.0       0  

[303 rows x 14 columns]

# Exercise 3: Creating new feature

1. Create a new feature 'AgeGroup' based on the 'age' column with the following labels: 'Young', 'Middle-aged', 'Senior' Learn how to use an int for bins parameter in the pd.cut() function to create the age groups.

2. Create a new feature 'HeartHealth' using the 'apply' function and the following conditions: Conditions: if thalach > 150 and chol < 240 Good, otherwise Poor

3. Create a new feature 'RiskFactor' using the following formula: RiskFactor = age * (chol + thalach)

# Exercise 4: Drawing plots with Pyplot module functions


1. Draw a line plot of the 'trestbps' column for the first 50 rows. Line Plot of Trend of Resting Blood Pressure. Creat the data that is needed for the plot by slicing the DataFrame to get the first 50 rows of the 'trestbps' column


2. Draw a line plot of average 'trestbps' for each 'age' group. Line plot of average Trend of Resting Blood Pressure. It illustrates the average trend of Resting Blood Pressure (trestbps) across different age groups.

- X-Axis: Age

- Y-Axis: Average Resting Blood Pressure


NOTE: Use groupby to group the data by 'age' and calculate the mean of 'trestbps'.


The plot provides insight into how the average Resting Blood Pressure changes with increasing age.




3. Draw a scatter plot showing the Relationship between Age and Risk Factor.


- X-Axis: Age

- Y-Axis: Risk Factor

4. Create a dataframe with the count of 'HeartHealth' for each 'AgeGroup' using the 'groupby' function. Learn about size() and unstack() functions.


5. Draw a stacked bar plot showing the HeartHealth distribution within each Age group as the x-axis and the 'Good' and 'Poor' categories stacked for each age group. Add a legend to the plot.


- X-Axis: Age Group

- Y-Axis: Count




6. Draw a histogram showing the distribution of patient ages. Set bins to 20 and color to 'skyblue'. Learn how to use the 'edgecolor' parameter to add a border to the bars.


- X-Axis: Age

- Y-Axis: Frequency (Number of individuals)




7. Draw a pie chart showing the percentage distribution of the 'AgeGroup' column. Learn how to use the 'autopct' parameter to display the percentage value and the 'startangle' parameter to rotate the pie chart.



- Each slice of the pie corresponds to a specific age group (Young, Middle-aged, Senior).

- The 'skyblue', 'orange', and 'lightgreen' colors distinguish between the age groups.

- The startangle parameter (90 degrees) adjusts the starting position of the first slice.




8. Draw a box plot showing the distribution of 'thalach' across different 'cp' values. Learn how to use the 'patch_artist' parameter to fill the boxes with color.


    This box plot visually represents the distribution of Maximum Heart Rate (thalach) for different chest pain types.

- X-Axis: Chest Pain Type

- Y-Axis: Maximum Heart Rate